### PR TITLE
use jquery-sortablejs instead of jquery-ui

### PR DESCRIPTION
**Pull request by dairidong**
use sortablejs to add a smoothy animation when reordering items.

### CODE REVIEW

**Comment by keriati** [comment]
Thanks for this patch. Would you be up to adjust it so that it doesn't need jquery at all and only uses sortable?

**Comment by dairidong** [comment]
> Thanks for this patch. Would you be up to adjust it so that it doesn't need jquery at all and only uses sortable?

Ok, I will do it later. Is it neccecary to restore the jquery-ui in the `webpack.mix.js`?

**Comment by keriati** [comment]
> > Thanks for this patch. Would you be up to adjust it so that it doesn't need jquery at all and only uses sortable?
> 
> 
> 
> Ok, I will do it later. Is it neccecary to restore the jquery-ui in the `webpack.mix.js`?

So the goal is to remove jquery and also jquery-ui from the code base step by step.

**Comment by dairidong** [comment]
> > > Thanks for this patch. Would you be up to adjust it so that it doesn't need jquery at all and only uses sortable?
> > 
> > 
> > Ok, I will do it later. Is it neccecary to restore the jquery-ui in the `webpack.mix.js`?
> 
> So the goal is to remove jquery and also jquery-ui from the code base step by step.

done.

**Comment by keriati** [comment]
Thanks. It looks mostly good. In Firefox when we use this new drag and drop it opens the items, so it is sadly not working correctly.
~~The file **resources/assets/js/jquery-ui.min.js** can also be removed with this PR I think.~~ seems like it is needed for autocomplete.

**Comment by keriati** [comment]
Add new Items is also broken with this PR, I think there is still a reliance on jquery-ui for autocomplete there.

**Comment by dairidong** [comment]
> Thanks. It looks mostly good. In Firefox when we use this new drag and drop it opens the items, so it is sadly not working correctly. ~The file **resources/assets/js/jquery-ui.min.js** can also be removed with this PR I think.~ seems like it is needed for autocomplete.

Ok, I will try to figure out the problem of Firefox, ~~if someday jquery-ui is removed, perhaps this pr can provide some help~~ I think I have misunderstood what you mean, forget it(my english is not good, sorry).

**Comment by dairidong** [comment]
> Thanks. It looks mostly good. In Firefox when we use this new drag and drop it opens the items, so it is sadly not working correctly. ~The file **resources/assets/js/jquery-ui.min.js** can also be removed with this PR I think.~ seems like it is needed for autocomplete.

I had found why the problem occurs from https://github.com/SortableJS/Sortable/issues/985, the way to solve it need to set the option `forceFallback` to `false`, in this case dragged element can't be styled (actually there is no real element exists).

File: resources/assets/js/app.js [hunk 1]
```diff
@@ -52,15 +52,33 @@ $.when($.ready).then(() => {
...
-    stop() {
-      const idsInOrder = $("#sortable").sortable("toArray", {
-        attribute: "data-id",
-      });
+  const sortableEl = document.getElementById("sortable");
+  // eslint-disable-next-line no-undef
+  const sortable = Sortable.create(sortableEl, {
+    disabled: true,
+    animation: 150,
+    forceFallback: !/Windows.+Firefox/.test(navigator.userAgent),
```
Comment (keriati) [inline resources/assets/js/app.js]: The fix works now only on windows, MacOS still broken.
However I tested with:

```suggestion
    forceFallback: !(navigator.userAgent.toLowerCase().indexOf("firefox") > -1),
```

And it works also on Mac.

File: resources/assets/js/app.js [hunk 2]
```diff
@@ -52,15 +52,33 @@ $.when($.ready).then(() => {
...
+    draggable: ".item-container",
+    onEnd() {
+      const idsInOrder = sortable.toArray();
       $.post(`${base}order`, { order: idsInOrder });
     },
   });
-  $("#sortable").sortable("disable");
+  // prevent Firefox drag behavior
+  if (/Windows.+Firefox/.test(navigator.userAgent)) {
```
Comment (keriati) [inline resources/assets/js/app.js]:
```suggestion
  if (navigator.userAgent.toLowerCase().indexOf("firefox") > -1) {
```

**Review by keriati — COMMENTED** [review]
Thanks for the fix. It works on Windows. On MacOS firefox is detected differently: Please adjust the firefox check and rebuild prod js.

**Comment by dairidong** [comment]
> Thanks for the fix. It works on Windows. On MacOS firefox is detected differently: Please adjust the firefox check and rebuild prod js.

Thanks for your help. I have seen the feature of firefox only be enabled in Windows someplace, probably it is changed now.

**Comment by dairidong** [comment]
found another bug, when the sortable element is not exists, some event will be wrong, I tried the `#test_config` button in the edit page and create page, it would save the form. 

 I had fixed it in the latest commit.

**Review by keriati — APPROVED** [review]
lgtm

**Pull request merged**